proof/infoflow/refine/ADT_IF_Refine.thy

```diff
@@ -784,6 +784,17 @@ lemma scheduler_if'_valid_duplicates[wp]:
   apply (wp hoare_drop_imps | simp)+
   done
 
+lemma nonzero_gt_zero[simp]:
+  "x \<noteq> (0 :: obj_ref) \<Longrightarrow> x > 0"
```

Comment:
This one is not a good global simp rule, because it will try to prove the antecedent every time the goal has a `0 < x` in it. Locally it may well make sense, though.